apps/website/blog/2025-09-05-maci-coordinator-service.md

```diff
@@ -0,0 +1,167 @@
+
+We chose to separate each action into its own endpoint to avoid long-running computations that could lead to network errors or timeouts.
+
+### Schedule
+
+After finishing up the finalisation module, we realized that it would be a lot easier for users to create polls and schedule the finalization process. This removes the need for users to manually call the finalization endpoints described above.
+
+1. To schedule a deployed poll you can use `https://coordinator.maci.vote/v1/scheduler/register`.
+2. To check if a poll is scheduled you can use `https://coordinator.maci.vote/v1/scheduler/status`
+3. To delete a scheduled poll you can use `https://coordinator.maci.vote/v1/scheduler/delete`
+
+## How to use
+
+### Run it from source
+
+1. Clone the MACI repository: [https://github.com/privacy-scaling-explorations/maci](https://github.com/privacy-scaling-explorations/maci)
+2. Install dependencies and build the project
+
+```bash
+pnpm install
+pnpm run build
+```
+
+3. Download the zkeys for the coordinator
+
+```bash
+pnpm run download-zkeys:ceremony:coordinator
+```
+
+4. Generate a coordinator MACI keypair. Remember to store those values for later.
+
+```bash
+pnpm run generate-maci-keypair
+```
+
+5. Move to the coordinator service directory
+
+```bash
+cd apps/coordinator
+```
+
+6. Configure your `.env` file. Make sure to configure it with your own secure values. The coordinator MACI private key from the previous step should be set here
+
+```bash
+cp .env.example .env
+```
+
+7. Start the service
+
+```bash
+pnpm run start
+# or
+pnpm run start:prod
+```
+
+### Run it with Docker
+
+1. Clone the MACI repository: [https://github.com/privacy-scaling-explorations/maci](https://github.com/privacy-scaling-explorations/maci)
+2. Go inside the project directory
+
+```bash
+cd maci
+```
+
+3. Build the docker image and run the container
+
+```bash
+# Build docker
+docker compose -f apps/coordinator/docker-compose.yml build
+
+# Run container detached
+docker compose -f apps/coordinator/docker-compose.yml up -d
+```
+
+## Usage notes
+
+1. You can check out the documentation while the service is running in [http://localhost:3001/api](http://localhost:3001/api)
+
+2. We have a [e2e.deploy.test.ts](https://github.com/privacy-scaling-explorations/maci/blob/main/apps/coordinator/tests/e2e.deploy.test.ts) file that performs a complete process: deployment, voting and finalization. You can use it to guide yourself.
+
+## Future work
+
+We are researching about different cryptographic methods to decentralize the coordinator. The idea is to eliminate reliance on a single centralized service for finalizing polls. A group of people would be able to spin up their own coordinator services and finalize the poll in a multi-party computation. For more info check out our MACI v4 research.
```

Comment:
I'm not sure the finalisation would necessarily be a multiparty computation like that, I kept it generic - let me know if you disagree. Also we don't have anything to point to yet for v4 so removed that sentence

```suggestion
We are researching about different cryptographic methods to decentralize the coordinator. The idea is to eliminate reliance on a single centralized entity for running MACI polls. Multiple entities would be able to spin up their own coordinator services and contribute to running a MACI poll in a decentralized way.
```